crates/bevy_ecs/src/system/builder.rs

```diff
@@ -337,6 +339,194 @@ unsafe impl<'s, T: FromWorld + Send + 'static> SystemParamBuilder<Local<'s, T>>
     }
 }
 
+/// A [`SystemParamBuilder`] for a [`FilteredResources`].
+/// See the [`FilteredResources`] docs for examples.
+pub struct FilteredResourcesParamBuilder<T>(T);
+
+impl<T> FilteredResourcesParamBuilder<T> {
+    /// Creates a [`SystemParamBuilder`] for a [`FilteredResources`] that accepts a callback to configure the [`FilteredResourcesBuilder`].
+    pub fn new(f: T) -> Self
+    where
+        T: FnOnce(&mut FilteredResourcesBuilder),
```

Comment:
Since `FilteredResourceBuilder` has a `.add_write` method it seems weird to give access to it in `FilteredResourcesParamBuilder`.

Actually, do we even need a readonly version of `FilteredResourceBuilder`? And I guess also `FilteredEntityRef` vs `FilteredEntityMut` seems a bit weird. After all the access already defines what can and can't be mutably borrowed, so what value does a static distinction provides?

Reply:
It didn't seem worth having a second builder type, but I could add it if you think that would be more clear.  

The advantage of the read-only versions is that they're `Clone` (and maybe even `Copy` if we can make `Access` into a reference), and that they `impl ReadOnlySystemParameter`.

Reply:
> It didn't seem worth having a second builder type, but I could add it if you think that would be more clear.

Yeah IMO it's confusing the fact that you can request write access on a read only `FilteredResourceBuilder`

> The advantage of the read-only versions is that they're `Clone` (and maybe even `Copy` if we can make `Access` into a reference), and that they `impl ReadOnlySystemParameter`.

I see, those are indeed nice advantages of a read-only version so it makes sense to keep the distinction.